scylla/src/client/session.rs

```diff
@@ -818,9 +818,6 @@ impl Session {
             identity: config.identity,
         };
 
-        #[cfg(feature = "metrics")]
-        let metrics = Arc::new(Metrics::new());
-
         let pool_config = PoolConfig {
             connection_config,
```

Comment:
Commit: `metrics: unpub Metrics::new() `

How is this change related to the commit? It removes the duplicate unused creation of `Metrics`.
This problem is not present in the code before this PR, so some other commit must introduce it, right?
In such case such older commit should be edited, instead of making this unrelated change in the older commit.

I suspect that clippy is failing because of such duplicate definition, right? If so, the checklist you filled in the PR (`All commits compile, pass static checks and pass test.`) is false.